src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNode.cs

```diff
@@ -0,0 +1,282 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+// MUX reference de78834
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class TreeViewNode : DependencyObject, ICustomPropertyProvider, IStringable
+	{
+		private WeakReference<TreeViewNode> m_parentNode;
+		private bool m_HasUnrealizedChildren;
+		private object m_itemsSource = null;
+		private ItemsSourceView m_itemsDataSource = null;
+
+		public TreeViewNode()
+		{
+			var collection = new TreeViewNodeVector();
+			collection.SetParent(this);
+			Children = collection;
+			collection.VectorChanged += ChildVectorChanged;
+		}
+
+		public TreeViewNode Parent
+		{
+			get
+			{
+				TreeViewNode parentNode = null;
+				m_parentNode?.TryGetTarget(out parentNode);
+				return parentNode;				
+			}
+
+			internal set
+			{
+				if (value != null)
+				{
+					m_parentNode = new WeakReference<TreeViewNode>(value);
+				}
+				else
+				{
+					m_parentNode = null;
+				}
+
+				//A parentless node has a depth of -1, and the first level of visible nodes in
+				// the tree view will have a depth of 0 (-1 + 1);
+				UpdateDepth(value != null ? value.Depth + 1 : -1);
+			}
+		}
+
+		public bool HasUnrealizedChildren
+		{
+			get => m_HasUnrealizedChildren;
+			set
+			{
+				m_HasUnrealizedChildren = value;
+				UpdateHasChildren();
+			}
+		}
+
+		public IList<TreeViewNode> Children { get; }
+
+		internal bool IsContentMode { get; set; }
+
+		internal TreeNodeSelectionState SelectionState { get; set; }
+
+		private void UpdateDepth(int depth)
+		{
+			// Update our depth
+			SetValue(DepthProperty, depth);
+
+			// Update children's depth
+			foreach (var child in Children)
+			{
+				child.UpdateDepth(depth + 1);
+			}
+		}
+
+		private void UpdateHasChildren()
+		{
+			bool hasChildren = Children.Count != 0 || m_HasUnrealizedChildren;
+			SetValue(HasChildrenProperty, hasChildren);
+		}
+
+		private void ChildVectorChanged(IObservableVector<TreeViewNode> sender, IVectorChangedEventArgs args)
+		{
+			var collectionChange = args.CollectionChange;
+			var index = args.Index;
+			UpdateHasChildren();
+			RaiseChildrenChanged(collectionChange, index);
+		}
+
+		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
```

Comment:
Codacy found an issue: [Remove the unused local variable 'property'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)